examples/experimental/dagster-airlift/dagster_airlift/core/load_defs.py

```diff
@@ -41,6 +47,123 @@
+    *,
+    airflow_instance: AirflowInstance,
+    defs: Optional[Definitions] = None,
+    sensor_minimum_interval_seconds: int = DEFAULT_AIRFLOW_SENSOR_INTERVAL_SECONDS,
+) -> Definitions:
+    defs = defs or Definitions()
+    # Probably a temp workaround?
+    if DefinitionsLoadContext.context_is_set():
+        context = DefinitionsLoadContext.get()
+    else:
+        context = DefinitionsLoadContext(load_type=DefinitionsLoadType.INITIALIZATION)
+    metadata_key = f"{AIRFLOW_SOURCE_METADATA_KEY_PREFIX}/{airflow_instance.name}"
+    definitions_metadata = _get_or_create_definitions_metadata(
+        context, metadata_key, airflow_instance, defs
+    )
+    return definitions_from_metadata(
+        context, definitions_metadata, defs, airflow_instance, sensor_minimum_interval_seconds
+    )
+
+
+def _get_or_create_definitions_metadata(
+    context: DefinitionsLoadContext,
+    metadata_key: str,
+    airflow_instance: AirflowInstance,
+    defs: Definitions,
+) -> "CacheableSpecs":
+    if (
+        context.load_type == DefinitionsLoadType.RECONSTRUCTION
+        and metadata_key in context.reconstruction_metadata
+    ):
+        return deserialize_value(context.reconstruction_metadata[metadata_key], CacheableSpecs)
+    else:
+        return compute_cacheable_assets(airflow_instance=airflow_instance, defs=defs)
+
+
+def compute_cacheable_assets(
+    airflow_instance: AirflowInstance, defs: Definitions
+) -> "CacheableSpecs":
+    migration_state = airflow_instance.get_migration_state()
+    dag_infos = {dag.dag_id: dag for dag in airflow_instance.list_dags()}
+    cacheable_task_data = construct_cacheable_assets_and_infer_dependencies(
+        definitions=defs,
+        migration_state=migration_state,
+        airflow_instance=airflow_instance,
+        dag_infos=dag_infos,
+    )
+
+    dag_specs_per_key: Dict[AssetKey, CacheableAssetSpec] = {}
+    for dag in dag_infos.values():
+        source_code = airflow_instance.get_dag_source_code(dag.metadata["file_token"])
+        dag_specs_per_key[dag.dag_asset_key] = get_cached_spec_for_dag(
+            airflow_instance=airflow_instance,
+            task_asset_keys_in_dag=cacheable_task_data.all_asset_keys_per_dag_id.get(
+                dag.dag_id, set()
+            ),
+            downstreams_asset_dependency_graph=cacheable_task_data.downstreams_asset_dependency_graph,
+            dag_info=dag,
+            source_code=source_code,
+        )
+    return CacheableSpecs(
+        dag_asset_specs=dag_specs_per_key,
+        task_asset_specs=cacheable_task_data.cacheable_specs_per_asset_key,
+    )
+
+
+def definitions_from_metadata(
+    context: DefinitionsLoadContext,
+    definitions_metadata: "CacheableSpecs",
+    defs: Definitions,
+    airflow_instance: AirflowInstance,
+    sensor_minimum_interval_seconds: int,
+) -> Definitions:
+    new_assets_defs = [
+        external_asset_from_spec(
+            dag_spec.to_asset_spec(additional_metadata={}),
```

Comment:
I'd like a convention where we suffix cacheable spes with `cacheable_spec`. A little more verbose but much more clear IMO.